### PR TITLE
Bugfix/Fix issue where user is not retrieved properly in the navbar on page reload

### PR DESCRIPTION
**Pull request by thilo-behnke**
### Checklist
- [x] I tested the changes and all related features on the test server https://artemistest.ase.in.tum.de.
- [x] I documented my source code using the JavaDoc / JSDoc style.
- [x] ~I added integration test cases for the server (Spring) related to the features~
- [x] I added integration test cases for the client (Jest) related to the features
- [x] I added screenshots/screencast of my UI changes
- [x] I translated all the newly inserted strings

### Description

Regression from https://github.com/ls1intum/Artemis/pull/694.

User does not see the overview page and the account info after reloading the page (see screenshot below). Pages are still accessible though if you know the URL OR by clicking on the Artemis logo in the header.

In detail:
The issue was that the authenticationState Subject in the `account.service.ts` was not a BehaviorSubject but a Subject. This way subscribers would not receive the most current user when subscribing but just nothing until the next authenticationState update (log in / log out).
I have also moved all the single emits in the service into a setter.
Also we now have a couple of tests for the account service.

In short:
- make authenticated subject in account service a behavior subject so the latest user value is emitted on subscribe
- correct typing for user subscription (can be null!)

### Steps for Testing

1. Log in to ArTEMiS
2. Reload the page
3. User should still be visible in the top right in account menu, overview should also still be available

### Screenshots

image

### CODE REVIEW

File: src/main/webapp/app/core/auth/account.service.ts [hunk 1]
```diff
@@ -21,15 +22,15 @@ export interface IAccountService {
...
     isAuthenticated: () => boolean;
-    getAuthenticationState: () => Observable<User>;
+    getAuthenticationState: () => Observable<User | null>;
     getImageUrl: () => string | null;
 }
 
 @Injectable({ providedIn: 'root' })
 export class AccountService implements IAccountService {
-    private userIdentity: User | null;
+    private userIdentityValue: User | null = null;
```
Comment (krusche) [inline src/main/webapp/app/core/auth/account.service.ts]: why do we need `null = null`?

Comment (thilo-behnke) [reply src/main/webapp/app/core/auth/account.service.ts]: This gives the userIdentity the inital value `null`, if I would not do this, the initial value would be `undefined`.
This would then represent a third state "has not logged in yet" besides "logged in" (`typeof User`) and "logged out" (`null`) that we have no use case for atm.

As the `authenticationState` subject should only emit changes (e.g. `null => User => null` vs. `null => null => User => null`), this would complicate the logic a bit as we would have to differentiate changes as `undefined => null => User => null` where the first two emits will not make a difference for the subscribers.

Comment (krusche) [reply src/main/webapp/app/core/auth/account.service.ts]: ok, I understand. It looks just weird how this is written. I'm not used to type definitions such as `User | null`. Could we not just write 
`private userIdentityValue: User? = null;`

or would this imply undefined as well?

Comment (krusche) [reply src/main/webapp/app/core/auth/account.service.ts]: I merge this now, we can talk about this on Tuesday
